### DISCUSSION TITLE
Defining a non-square field

### DISCUSSION THREAD

**Discussion by naninano1** [e0404/matRad#576] [Q&A]
Hi,
In order to verify the dose distribution (PDD and profile) calculation of matRad against Monte Carlo, I want to define a non-square static field, e.g. a triangle field. I obtained the PDD and profiles for different square-shaped fields by defining a cubic water phantom and "bixelWidths =" command. But I am not sure if it is possible to do it for a non-square field or not.

**Comment by wahln** [comment]
Hi @naninano1,

sorry for the delay, for some reasons I didn't get notifications on the discussions.
You can also use non-square fields with the pencil-beam dose calculation, but it is a little bit of hacking in the matRad_generateStf (if your triangle field should be discretized in square bixels). If you want a pure triangle field, it should also be possible by creating a single ray and hacking the triangle field into the dose calculation. - there, the fluence (which you could hardcode to be triangular) is convolved with the pencil-beam kernel.
